### PR TITLE
add_users_endpoints

### PR DESCRIPTION
**Pull request by Cheephot**

### CODE REVIEW

File: src/MessageService.Business/Commands/Channels/GetChannelCommand.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      _rcGetUsers = rcGetUsers;
+      _logger = logger;
+      _responseCreator = responseCreator;
+    }
+
+    public async Task<OperationResultResponse<ChannelInfo>> ExecuteAsync(Guid channelId)
+    {
+      DbChannel dbChannel = await _channelRepository.GetAsync(channelId);
+
+      if (dbChannel is null)
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Channels/GetChannelCommand.cs:85]: why you don't check rights? (or including in channel)

File: src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      if (!validationResult.IsValid)
+      {
+        return _responseCreater.CreateFailureResponse<bool>(HttpStatusCode.BadRequest);
+      }
+
+      OperationResultResponse<bool> response = new();
+
+      await _repository.RemoveAsync(channelId, usersIds);
+
+      Guid createdBy = _httpContextAccessor.HttpContext.GetUserId();
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs:66]: you don't use it

File: src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+      response.Body = await _repository.CreateAsync(
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs:68]: isAdmin is always true

File: src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  public class CreateWorkspaceUserCommand : ICreateWorkspaceUserCommand
+  {
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IAccessValidator _accessValidator;
+    private readonly ICreateWorkspaceUserValidator _validator;
+    private readonly IDbWorkspaceUserMapper _mapper;
+    private readonly IWorkspaceUserRepository _repository;
+    private readonly IResponseCreater _responseCreater;
+
+    public CreateWorkspaceUserCommand(
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs:29]: mb UserS

File: src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      _accessValidator = accessValidator;
+      _validator = validator;
+      _mapper = mapper;
+      _repository = repository;
+      _responseCreater = responseCreater;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid workspaceId, List<Guid> usersIds)
+    {
+      bool isAdmin = true;
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs:47]: same

File: src/MessageService.Business/Commands/User/RemoveChannelUserCommand.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      IChannelUserRepository repository,
+      IAccessValidator accessValidator,
+      IResponseCreater responseCreater)
+    {
+      _repository = repository;
+      _accessValidator = accessValidator;
+      _responseCreater = responseCreater;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid channelId, List<Guid> usersIds)
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/User/RemoveChannelUserCommand.cs:31]: I would combine it into a model and validate it not here

File: src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      }
+
+      OperationResultResponse<bool> response = new();
+
+      response.Body = await _repository.RemoveAsync(workspaceId, usersIds);
+      response.Status = OperationResultStatusType.FullSuccess;
+
+      if (response.Body)
+      {
+        response = _responseCreater.CreateFailureResponse<bool>(HttpStatusCode.NotFound);
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs:50]: return

File: src/MessageService.Data/ChannelUserRepository.cs [hunk 8]
```diff
@@ -1,24 +1,70 @@
...
+
       _provider.ChannelsUsers.AddRange(dbChannelUsers);
       await _provider.SaveAsync();
+
+      return true;
+    }
+
+    public async Task<DbChannelUser> GetAsync(Guid userId, Guid channelId)
+    {
+      return await _provider.ChannelsUsers
```
Comment (BelehovEgor) [inline src/MessageService.Data/ChannelUserRepository.cs:42]: do you need to check isActive?

File: src/MessageService.Business/Commands/Channels/GetChannelCommand.cs (hunk 1 above)
Comment (MariyaSin) [reply src/MessageService.Business/Commands/Channels/GetChannelCommand.cs:85]: Only channel users can get channel

File: src/MessageService.Business/Commands/Channels/GetChannelCommand.cs [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+      {
+        return _responseCreator.CreateFailureResponse<ChannelInfo>(HttpStatusCode.NotFound);
+      }
+
+      OperationResultResponse<ChannelInfo> response = new();
+
+      List<UserData> usersData = await GetUsersAsync(dbChannel.Users?.Select(u => u.WorkspaceUserId).ToList(), response.Errors);
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/Channels/GetChannelCommand.cs:92]: you select workspaceUsersId, workspaceUsersId != userId

File: src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+      _accessValidator = accessValidator;
+      _validator = validator;
+      _mapper = mapper;
+      _repository = repository;
+      _responseCreater = responseCreater;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid channelId, List<Guid> usersIds)
+    {
+      bool isAdmin = true;
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs:47]: what for do you create the  variable?

File: src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+      if (!await _accessValidator.IsAdminAsync())
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs:49]: you check what user is application admin, why?

File: src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+      response.Body = await _repository.CreateAsync(
+        usersIds.Select(userId => _mapper.Map(channelId, userId, isAdmin, _httpContextAccessor.HttpContext.GetUserId())).ToList());
+
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+
+      response.Status = OperationResultStatusType.FullSuccess;
+
+      if (!response.Body)
+      {
+        return _responseCreater.CreateFailureResponse<bool>(HttpStatusCode.BadRequest);
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs:77]: you use return twice

File: src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+      if (!await _accessValidator.IsAdminAsync())
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs:49]: same

File: src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+      ValidationResult validationResult = await _validator.ValidateAsync(usersIds);
+
+      if (!validationResult.IsValid)
+      {
+        return _responseCreater.CreateFailureResponse<bool>(HttpStatusCode.BadRequest);
+      }
+
+      OperationResultResponse<bool> response = new();
+
+      await _repository.RemoveAsync(workspaceId, usersIds);
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs:64]: what for do you do it?

File: src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+      response.Body = await _repository.CreateAsync(
+        usersIds.Select(userId => _mapper.Map(userId, workspaceId, isAdmin, _httpContextAccessor.HttpContext.GetUserId())).ToList());
+
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+
+      response.Status = OperationResultStatusType.FullSuccess;
+
+      if (!response.Body)
+      {
+        return _responseCreater.CreateFailureResponse<bool>(HttpStatusCode.BadRequest);
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/CreateWorkspaceUserCommand.cs:75]: same

File: src/MessageService.Business/Commands/User/RemoveChannelUserCommand.cs [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+    {
+      if (!await _accessValidator.IsAdminAsync())
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/RemoveChannelUserCommand.cs:33]: same

File: src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs [hunk 17]
```diff
@@ -0,0 +1,56 @@
+﻿using System;
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs:1]: RemoveWorkspaceUsersCommand

File: src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+using LT.DigitalOffice.Kernel.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Enums;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.MessageService.Business.Commands.User.Interfaces;
+using LT.DigitalOffice.MessageService.Data.Interfaces;
+
+namespace LT.DigitalOffice.MessageService.Business.Commands.User
+{
+  public class RemoveWorkspaceUserCommand : IRemoveWorkspaceUserCommand
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs:15]: RemoveWorkspaceUsersCommand

File: src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+      IResponseCreater responseCreater)
+    {
+      _repository = repository;
+      _accessValidator = accessValidator;
+      _responseCreater = responseCreater;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid workspaceId, List<Guid> usersIds)
+    {
+      if (!await _accessValidator.IsAdminAsync())
```
Comment (MariyaSin) [inline src/MessageService.Business/Commands/User/RemoveWorkspaceUserCommand.cs:33]: same

File: src/MessageService.Data/ChannelUserRepository.cs [hunk 20]
```diff
@@ -1,24 +1,70 @@
...
+        .FirstOrDefaultAsync(x => x.ChannelId == channelId && x.WorkspaceUserId == userId);
+    }
+
+    public async Task<bool> RemoveAsync(Guid channelId, IEnumerable<Guid> usersIds)
+    {
+      List<DbChannelUser> dbChannelUsers = await _provider.ChannelsUsers
+        .Where(cu => cu.IsActive && cu.ChannelId == channelId && usersIds.Contains(cu.WorkspaceUserId))
```
Comment (MariyaSin) [inline src/MessageService.Data/ChannelUserRepository.cs:49]: are you shure userId == WorkspaceUserId

File: src/MessageService.Mappers/Models/ChannelInfoMapper.cs [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+    {
+      if (dbChannel is null)
+      {
+        return null;
+      }
+
+      return new ChannelInfo
+      {
+        Id = dbChannel.Id,
+        Avatar = new ImageConsist()
```
Comment (MariyaSin) [inline src/MessageService.Mappers/Models/ChannelInfoMapper.cs:31]: return null if not image content

File: src/MessageService.Mappers/Models/ChannelInfoMapper.cs [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+        {
+          Content = dbChannel.ImageContent,
+          Extension = dbChannel.ImageExtension
+        },
+        Name = dbChannel.Name,
+        IsActive = dbChannel.IsActive,
+        IsPrivate = dbChannel.IsPrivate,
```
Comment (MariyaSin) [inline src/MessageService.Mappers/Models/ChannelInfoMapper.cs:38]: only users in private channel can see the channel

File: src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs (hunk 11 above)
Comment (Cheephot) [reply src/MessageService.Business/Commands/User/CreateChannelUserCommand.cs:49]: I thought сreate channel has right only admin, or all workspace user has right create channel?

**Pull request closed without merging**